drivers/hmis/app/graphql/mutations/create_enrollment.rb

```diff
@@ -0,0 +1,61 @@
+module Mutations
+  class CreateEnrollment < BaseMutation
+    argument :input, Types::HmisSchema::CreateEnrollmentValues, required: true
+
+    field :enrollments, [Types::HmisSchema::Enrollment], null: true
+    field :errors, [Types::HmisSchema::ErrorGroup], null: false
+
+    def resolve(input:)
+      user = hmis_user
+      errors = []
+
+      if input.household_members.select { |hm| hm.relationship_to_ho_h == 1 }.size != 1
+        return {
+          enrollments: [],
+          errors: [
+            {
+              id: 'input',
+              errors: [
+                ArgumentError.new('Exactly one client must be head of household'),
+              ],
+            },
+          ],
+        }
+      end
+
+      if Date.parse(input.start_date) > Date.today
+        return {
+          enrollments: [],
+          errors: [
+            {
+              id: 'input',
+              errors: [
+                ArgumentError.new('Entry date cannot be in the future'),
+              ],
+            },
+          ],
+        }
+      end
+
+      enrollments = input.to_enrollments_params.map do |attrs|
+        enrollment = Hmis::Hud::Enrollment.new(data_source_id: user.data_source_id, **attrs)
+
+        if enrollment.valid? && !enrollment.in_progress?
+          enrollment.save!
+        else
+          enrollment.in_progress = true
+          enrollment.save(validate: false)
+        end
+
+        errors << { id: enrollment.id, class_name: enrollment.class.name, errors: enrollment.errors } unless enrollment.errors.empty?
+
+        enrollment
+      end
+
+      {
+        enrollments: enrollments,
+        errors: errors,
```

Comment:
I'm not sure how I feel about the shape of `errors` changing. Maybe instead of this, we could attach the `id` to each error inside `enrollment.errors`?  I could be convinced either way , I think.

Reply:
Yeah, in retrospect I think I'd prefer to have an ID in each error. The tricky bit is that the actual objects backing the `ValidationError` type are `ActiveModel::Error` objects, so I can't just append an ID key to them and call it a day. But I'll take another look and see if I can change it.

Reply:
I decided to flatten this and add an ID field to ValidationError, but I still need to get the actual enrollment IDs in there. We might need to think a bit about a pattern for that, since we're passing actual `Error` objects into `errors`

Reply:
Tweaked this a bit to add a custom error type that takes whatever kwargs you give it. Probably not a perfect solution, but it's a start.